modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-sets/FieldSetModal.js

```diff
@@ -0,0 +1,472 @@
+import {errorToast, successToast} from '../../../js/utils/toast.es';
+import {FormBuilder} from '../../FormBuilder';
+import {EVENT_TYPES} from '../../eventTypes';
+import sidebarReducer from '../../reducers/sidebarReducer';
+
+const DEFAULT_DATA_DEFINITION = {
+	dataDefinitionFields: [],
+};
+
+const DEFAULT_DATA_LAYOUT = {
+	dataLayoutFields: {},
+	dataLayoutPages: [],
+	dataRules: [],
+	paginationMode: 'single-page',
+};
+
+const ModalContent = ({
+	fieldSet: fieldSetProp,
+	onClose,
+	onUpdate,
+	updateFieldSetList,
+}) => {
+	const dispatch = useForm();
+	const {contentType} = useConfig();
+	const {
+		dataDefinitionId,
+		defaultLanguageId,
+		editingLanguageId,
+	} = useFormState();
+	const {dataDefinition, dataLayout} = useFormState({
+		schema: ['dataDefinition', 'dataLayout'],
+	});
+
+	const [name, setName] = useState(dataDefinition.name ?? {});
+	const changeEditingLanguageId = (editingLanguageId) =>
+		dispatch({
+			payload: {editingLanguageId},
+			type: EVENT_TYPES_CORE.LANGUAGE.CHANGE,
+		});
+
+	const onSave = async () => {
+		const {
+			availableLanguageIds,
+			dataDefinitionFields,
+		} = dataDefinition.build();
+
+		const {dataLayoutPages} = dataLayout.build();
+
+		try {
+			if (!name[defaultLanguageId]) {
+				changeEditingLanguageId(defaultLanguageId);
+				throw new Error(
+					Liferay.Language.get('please-enter-a-valid-title')
+				);
+			}
+
+			const fieldsWithoutOptions = getFieldsWithoutOptions(
+				dataDefinitionFields,
+				defaultLanguageId
+			);
+
+			if (fieldsWithoutOptions.length) {
+				throw new Error(
+					Liferay.Util.sub(
+						Liferay.Language.get(
+							'at-least-one-option-should-be-set-for-field-x'
+						),
+						fieldsWithoutOptions[0].label[defaultLanguageId]
+					)
+				);
+			}
+
+			if (dataDefinitionId) {
+
+				// update fieldSet
+
+				const updatedFieldSet = {
+					...fieldSetProp,
+					availableLanguageIds,
+					dataDefinitionFields,
+					defaultDataLayout: normalizeDataLayout(
+						{
+							...fieldSetProp.defaultDataLayout,
+
+							dataLayoutPages: dataLayoutPages.map((page) => {
+								return {
+									...page,
+									description: {
+										[defaultLanguageId]: '',
+									},
+
+									title: {
+										[defaultLanguageId]: '',
+									},
+								};
+							}),
+
+							name,
+						},
+						defaultLanguageId
+					),
+					name,
+				};
+
+				const onPropagate = async () => {
+					const savedFieldset = await updateItem(
+						`/o/data-engine/v2.0/data-definitions/${dataDefinitionId}`,
+						updatedFieldSet
+					);
+
+					updateFieldSetList({fieldSet: savedFieldset});
+					successToast(Liferay.Language.get('fieldset-saved'));
+				};
+
+				onClose();
+				onUpdate({onPropagate, updatedFieldSet});
+			}
+			else {
+
+				// create new fieldSet from scratch
+
+				const fieldSet = {
+					availableLanguageIds,
+					dataDefinitionFields,
+					defaultDataLayout: {
+						dataLayoutPages,
+						name,
+					},
```

Comment:
you can get the `dataLayout` structure here.

```suggestion
					defaultDataLayout: dataLayout,
```